### DISCUSSION TITLE
Offline logging: Progressive Web App?

### DISCUSSION THREAD

**Discussion by Cadair** [magicbug/Cloudlog#1482] [Ideas / Feature Requests]
This is a little mad, and a big ask, but something I have been thinking about this week.

I use cloudlog hosted on a server in my house and I love it. It's easy to use and is always where I am on any device. Until that is I go hiking up a hill somewhere and have no phone signal.

It would be *really cool* if the frontend could be a progressive web app which let you log to local browser storage until a connection to the server was re-established and then synced with the database. This would also give a more native feeling experience on android where you can "install" PWAs to the home screen etc.

**Comment by phl0** [comment]
Maybe this is an option?

https://github.com/myzinsky/cloudLogOffline

**Reply by Cadair** [reply]
Looks cool, thanks for the link.

**Reply by Werzi2001** [reply]
@phl0 Do you use the app? I was thinking about trying to use it for SOTA activations without internet access. Does it work as described/stable? I'm a bit worried as the last update is more than a year ago. What about newer features like the log books?

**Reply by phl0** [reply]
> @phl0 Do you use the app? I was thinking about trying to use it for SOTA activations without internet access. Does it work as described/stable? I'm a bit worried as the last update is more than a year ago. What about newer features like the log books?

Jepp I am using it. A few days ago I added support for station_logbooks aka. V2 of Cloudlog. See https://github.com/myzinsky/cloudLogOffline/pull/51. 
And @myzinsky has just released v1.1.0 which I am currently waiting for in the Android app store: https://github.com/myzinsky/cloudLogOffline/releases/tag/1.1.0

**Reply by Werzi2001** [reply]
@phl0 Thanks a lot for the input. Great to hear the project is not "dead". I'll give it a try then.

**Reply by phl0** [reply]
Sure. The author is very friendly and I am sure he will accept PRs with good content :)

**Reply by myzinsky** [reply]
I designed the app for the SOTA and COTA usage explicitly ;-)

**Comment by magicbug** [comment]
Offline Cloudlog has come up a lot and agree it would be nice to have, not a 100% sure how to implement it however

**Comment by HB9HIL** [comment]
I created a Feature Request for that

https://github.com/magicbug/Cloudlog/issues/2294

**Comment by HB9HIL** [comment]
I created a PR for the "App Behaviour"

Not Offline for now but maybe a start

https://github.com/magicbug/Cloudlog/pull/2654

**Reply by magicbug** [reply]
Merged into dev branch works great

**Comment by HB9HIL** [comment]
@Cadair 

The new Version 2.5.1 was just released! It now supports acting like a Progressive Web App but for now just online-only. 

For iPhones:

Tap on the Share Button in Safari and add Cloudlog to your Homescreen. It should act like a App and will be shown in Fullscreen.

On Android Phones you can choose "Install Cloudlog" from the Browser Menu. 

Same on Desktop Browsers (only Chrome and Edge, Safari and Firefox do not support PWA). On the right side of the URL Line should now appear the Option "install". 

Try that and give some feedback.
